coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/JoinSqlCallTransformer.java

```diff
@@ -75,24 +81,41 @@ private static boolean isUnnestOperatorPresentInRightSqlNode(SqlNode rightSqlNod
         && ((SqlCall) rightSqlNode).operand(0).getKind() == SqlKind.UNNEST;
   }
 
-  private static boolean isUnnestSqlCallCorrelated(SqlNode sqlNode) {
-    SqlNode aliasOperand = ((SqlCall) sqlNode).operand(0); //  unnest(x)
-    SqlNode unnestOperand = ((SqlCall) aliasOperand).operand(0); // x
+  /**
+   * Given an Unnest SqlCall, UNNEST('x'), the SqlCall is considered correlated when the unnest operand, 'x':
+   *      (1) References a column from the base tables:
+   *      Sample SqlCalls:
+   *          - UNNEST(table1.col)
+   *          - UNNEST(if(table1.col IS NOT NULL AND CAST(CARDINALITY(table1.col) AS INTEGER) > 0 table1.col, ARRAY[NULL] WITH ORDINALITY))
+   *          - UNNEST(split(table1.stringCol, "delimiter"))
+   *      (2) SqlCall with "TRANSFORM" operator to support unnesting array of structs
+   *      Sample SqlCalls:
+   *          - UNNEST(TRANSFORM(table1.col, x -> ROW(x)))
+   * @param unnestSqlCall unnest sqlCall
+   * @return true if the sqlCall is correlated to the outer query
+   */
+  private static boolean isUnnestSqlCallCorrelated(SqlCall unnestSqlCall) {
+    return isSqlCallCorrelation(unnestSqlCall);
+  }
 
-    // When the unnest operand, 'x', is:
-    // (1) SqlIdentifier referring to a column, ex: table1.col1
-    // (2) SqlCall with "IF" operator for outer unnest
-    // (3) SqlCall with "TRANSFORM" operator to support unnesting array of structs
-    // Substitute JoinType with CROSS JoinType.
-    if (unnestOperand.getKind() == SqlKind.IDENTIFIER
-        || (unnestOperand instanceof SqlCall
-            && ((SqlCall) unnestOperand).getOperator().getName().equalsIgnoreCase("transform"))
-        || (unnestOperand instanceof SqlCall
-            && ((SqlCall) unnestOperand).getOperator().getName().equalsIgnoreCase("if"))) {
-      return true;
+  private static boolean isSqlCallCorrelation(SqlCall sqlCall) {
+    for (SqlNode operand : sqlCall.getOperandList()) {
+      if (operand instanceof SqlIdentifier) {
```

Comment:
How do we know for sure that any identifier operand of an unnest operator references a column from the base tables?  Could it be a reference to a non-base table column?

Reply:
all tables referenced can be seen as base tables.